src/main/java/ua/softserveinc/tc/dao/UserDao.java

```diff
@@ -1,17 +1,20 @@
 package ua.softserveinc.tc.dao;
 
 import ua.softserveinc.tc.entity.Role;
+import ua.softserveinc.tc.entity.Room;
 import ua.softserveinc.tc.entity.User;
 
+import java.util.Date;
 import java.util.List;
 
 public interface UserDao extends BaseDao<User> {
-
     void deleteUserById(Long id);
 
     List<User> findAllUsersByRole(Role role) ;
 
     User getUserByEmail(String email);
 
     List<User> findAll(List<Long> ids);
-}
\ No newline at end of file
+
+    List<User> findActiveUsers(Date startDate, Date endDate, Room room);
```

Comment:
This method was extracted from ```UserService``` (as well as several others from different Services) because it was making database calls, which is responsibility of dao layer.